code/datums/components/paintable.dm

```diff
@@ -0,0 +1,31 @@
+/datum/component/spraycan_paintable
+    var/current_paint
+
+/datum/component/spraycan_paintable/Initialize()
+    RegisterSignal(COMSIG_PARENT_ATTACKBY, .proc/Repaint)
+
+/datum/component/spraycan_paintable/Destroy()
+    RemoveCurrentCoat()
+    return ..()
+
+/datum/component/spraycan_paintable/proc/RemoveCurrentCoat()
+    var/atom/A = parent
+    A.remove_atom_colour(FIXED_COLOUR_PRIORITY, current_paint)
+
+/datum/component/spraycan_paintable/proc/Repaint(obj/item/toy/crayon/spraycan/spraycan, mob/living/user)
+    if(!istype(spraycan) || user.a_intent == INTENT_HARM)
+        return FALSE
+    . = TRUE
+    if(spraycan.is_capped)
+        to_chat(user, "<span class='warning'>Take the cap off first!</span>")
+        return
+    if(spraycan.check_empty(user))
+        return
+    RemoveCurrentCoat()
+    spraycan.use_charges(2)
```

Comment:
You don't actually check if the can has 2 charges. You should make it so that check_empty has a configurable amount to check for